src/components/ArticleVoteBtn.jsx

```diff
@@ -1,40 +1,69 @@
-import React from 'react'
+import { patchArticle } from "../lib/api";
+import { useState } from "react";
 
-const ArticleVoteBtn = ({votes}) => {
+const ArticleVoteBtn = (props) => {
+  const { article, setArticle } = props;
+  const [voteChange, setVoteChange] = useState(0);
+  const [error, setError] = useState(null);
+  const handleVote = (vote) => {
+    setVoteChange(curr=>curr+vote);
+    patchArticle(article.article_id, vote)
+      .then(() => {
+        console.log(success);
+      })
+      .catch(() => setError(true));
+  };
+  let voteDisplay= article.votes + voteChange
+  if(error){
```

Comment:
the best place to handle this is within the .catch(), there we can set the vote state back to the original number of votes